mathphys/functions.py

```diff
@@ -89,3 +92,70 @@ def load_pickle(fname):
     with open(fname, 'rb') as fil:
         data = _pickle.load(fil)
     return data
+
+
+def repo_info(repo_path):
+    """Get repository information.
+
+    Args:
+        repo_path (str): Repository path.
+
+    Returns:
+        repo_info (dict): Repository info.
+            path, active_branch, last_tag, last_commit, is_dirty.
+
+    """
+    info = dict()
+    err = _subprocess.CalledProcessError
+    out = _subprocess.check_output
+    try:
+        cmd = ['git', '-C', repo_path, 'rev-parse', '--abbrev-ref', 'HEAD']
+        active_branch = out(cmd, universal_newlines=True).strip()
+    except err:
+        print(f'Repository path not found: {repo_path}.')
+        return info
+    try:
+        cmd = ['git', '-C', repo_path, 'describe', '--tags', '--abbrev=0']
+        last_tag = out(cmd, universal_newlines=True).strip()
+    except err:
+        last_tag = ''
+    cmd = ['git', '-C', repo_path, 'rev-parse', 'HEAD']
+    last_commit = out(cmd, universal_newlines=True).strip()[:7]
+    cmd = ['git', '-C', repo_path, 'status', '--short']
+    is_dirty = out(cmd, universal_newlines=True).strip() != ''
+    info['path'] = repo_path
+    info['active_branch'] = active_branch
+    info['last_tag'] = last_tag
```

Comment:
I think something is missing here. We need to know the hash of the commit of the tag, so that we know whether we are at the tag commit or if more commits were added after that.

Reply:
Right! Now I included the key `last_tag_commit` with that information. Thanks!